crates/loader/src/local/mod.rs

```diff
@@ -122,13 +121,11 @@ fn error_on_duplicate_ids(components: Vec<RawComponentManifest>) -> Result<()> {
 
 /// Validate fields in raw app manifest
 pub fn validate_raw_app_manifest(raw: &RawAppManifestAnyVersion) -> Result<()> {
-    match raw {
-        RawAppManifestAnyVersion::V1(raw) => {
-            raw.components
-                .iter()
-                .try_for_each(|c| validate_allowed_http_hosts(&c.wasm.allowed_http_hosts))?;
-        }
-    }
+    let manifest = raw.try_as_v1().expect("expected v1 manifest");
```

Comment:
Similar comment about compile time vs run time detection, though this one will of course be super obvious.

I wonder if an alternative approach would be to have _yet another_ enum a la `CanonicalisedRawAppManifestAnyVersion { V1(RawAppManifestV1) }` (Nobel Prize for Naming committee, take note), and the loader throws away the New and Old cases _right at the start_ and the rest of the program only ever sees the canonicalised one and can pattern match on it and _know_ at compile time that there is no `V2` case to consider.